script/index.js

```diff
@@ -0,0 +1,298 @@
+  div2.appendChild(h2);
+  div2.appendChild(p);
+  ul.appendChild(li1);
+  ul.appendChild(li2);
+  ul.appendChild(li3);
+  div2.appendChild(ul);
+  div1.appendChild(div2);
+  div1.appendChild(button);
+  myworks.appendChild(div1);
+
+  button.addEventListener('click', () => {
+    const trackkeeping = document.querySelector('.trackkeeping');
+    trackkeeping.textContent = list.title;
+
+    const trackkeepingDesk = document.querySelector('.desktexts');
+    trackkeepingDesk.textContent = list.title;
+
+    const desknewLangDesk = document.querySelectorAll('.DdesknewLang');
+    desknewLangDesk[0].textContent = list.languages1;
+    desknewLangDesk[1].textContent = list.languages2;
+    desknewLangDesk[2].textContent = list.languages3;
+
+    const desknewLang = document.querySelectorAll('.desknewLang');
+    desknewLang[0].textContent = list.languages1;
+    desknewLang[1].textContent = list.languages2;
+    desknewLang[2].textContent = list.languages3;
+
+    const imgpopup = document.querySelector('.img-popup');
+    imgpopup.src = list.image;
+
+    const imgpopupDesk = document.querySelector('.desktimg-popup');
+    imgpopupDesk.src = list.deskImage;
+
+    const thetxt = document.querySelector('.thetxt');
+    thetxt.textContent = list.description;
+
+    const thetxtDesk = document.querySelector('.desktthetxt');
+    thetxtDesk.textContent = list.deskDescription;
+
+    document.querySelector('.desktpopupcontainer').classList.toggle('expanded');
+    document.querySelector('.popupcontainer').classList.toggle('extended');
+  });
+});
+
+// form validation
+function valid() {
+  const email = document.getElementById('email').value;
+  const validatedEmail = email.toLowerCase();
+  if (validatedEmail === email) {
+    document.querySelector('.error').textContent = '';
+    return true;
+  }
+  document.querySelector('.error').textContent = 'Email address should always be in Lowercase.';
+  return false;
+}
+
+// stop button default action
+const vForm = document.querySelector('form');
+vForm.addEventListener('submit', (event) => {
+  if (valid() === false) {
+    event.preventDefault();
+  }
+});
+
+// local storage
+const form = document.getElementById('form');
+
+const nameInput = document.getElementsByName('user-name');
+const emailInput = document.getElementsByName('e-mail');
+const messageInput = document.getElementsByName('message');
+
+if (localStorage.getItem('form-data')) {
+  const formData = JSON.parse(localStorage.getItem('form-data'));
+  nameInput[0].value = formData.nameInputValue;
+  emailInput[0].value = formData.emailInputValue;
+  messageInput[0].value = formData.messageInputValue;
+}
+
+form.addEventListener('submit', (event) => {
+  event.preventDefault();
+  const formData = {
+    nameInputValue: nameInput[0].value,
+    emailInputValue: emailInput[0].value,
+    messageInputValue: messageInput[0].value,
+  };
+  localStorage.setItem('form-data', JSON.stringify(formData));
+  form.submit();
+});
```

Comment:
Good job so far 💯 but when the user changes the content of any input field in the form the data object is updated and it is saved to the local storage, for now, you're only doing it once the `form` is submitted.

To implement this change, you can modify the code to listen for changes on the input fields and update the `formData` object and local storage each time a change occurs. Here's an example implementation:
```
function updateFormData() {
  formData.nameInputValue = nameInput[0].value;
  formData.emailInputValue = emailInput[0].value;
  formData.messageInputValue = messageInput[0].value;
  localStorage.setItem('form-data', JSON.stringify(formData));
}

nameInput[0].addEventListener('input', updateFormData);
emailInput[0].addEventListener('input', updateFormData);
messageInput[0].addEventListener('input', updateFormData);
```
In this implementation, the `formData` object is initialized with empty values for each input field. Then, if there is saved form data in local storage, it is loaded and used to pre-fill the input fields. 💯 

The `updateFormData()` function is defined to update the `formData` object with the current values of each input field and save it to local storage. This function is then attached as an event listener to each input field using the `addEventListener()` method. This ensures that each time the user changes the content of any input field, the `formData` object is updated and saved to local storage. 🤔 

Overall, this implementation should provide a more responsive and reliable user experience for the form. 🤓